tests/static-analysis/unwrap.php

```diff
@@ -8,27 +8,71 @@
 use loophp\collection\Contract\Collection as CollectionInterface;
 
 /**
- * @param CollectionInterface<int, int> $collection
+ * @psalm-param CollectionInterface<int<0, 2>, 1|2|3> $collection
+ *
+ * @phpstan-param CollectionInterface<int, int> $collection
  */
-function unwrap_checkList(CollectionInterface $collection): void
+function unwrap_checkList1(CollectionInterface $collection): void
 {
 }
 /**
- * @param CollectionInterface<string, string> $collection
+ * @psalm-param CollectionInterface<int<0, 2>, 2|3|list{1}> $collection
```

Comment:
this is not good, and it shows that the new type hints have made `unwrap` too restrictive and incorrect in this case

```PHP
Collection::fromIterable([[1], 2, 3])->unwrap() 
// will result in Collection(1, 2, 3) aka `Collection<int, int>`, but the type hints think the resulting types are the same 
// as were passed in the original collection
```